client/my-sites/customer-home/locations/primary/index.jsx

```diff
@@ -17,14 +17,20 @@ const cardComponents = {
 };
 
 const Primary = ( { checklistMode, cards } ) => {
+	// Always ensure we have primary content.
+	if ( cards && cards.length < 1 ) {
+		cards = [ 'home-primary-quick-links' ];
```

Comment:
Non blocking but we might want to add logging here, either on the server or client when we hit this case.